shazam/shazam.py

```
@@ -0,0 +1,174 @@
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url, timeout=120.0) as response:
+                    response.raise_for_status()
+                    return await response.read()
+        except aiohttp.ClientError as error:
+            logging.exception("Error fetching media from URL: %s", url, exc_info=error)
+            raise commands.UserFeedbackCheckFailure("Failed to fetch media from the URL.") from error
+
+    async def get_dominant_color(self, image_url: str) -> discord.Color:
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(image_url) as response:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)